mne/decoding/tests/test_xdawn.py

```diff
@@ -0,0 +1,99 @@
+import numpy as np
+import os.path as op
+from nose.tools import (assert_raises)
+from numpy.testing import assert_array_equal
+from mne import (io, Epochs, read_events, pick_types,
+                 compute_raw_covariance)
+from mne.utils import requires_sklearn, run_tests_if_main
+from mne.decoding import XdawnTransformer
+from mne.preprocessing import Xdawn
+
+base_dir = op.join(op.dirname(__file__), '..', '..', 'io', 'tests', 'data')
+raw_fname = op.join(base_dir, 'test_raw.fif')
+event_name = op.join(base_dir, 'test-eve.fif')
+evoked_nf_name = op.join(base_dir, 'test-nf-ave.fif')
+
+tmin, tmax = -0.1, 0.2
+event_id = dict(cond2=2, cond3=3)
+
+
+def _get_data():
+    raw = io.read_raw_fif(raw_fname, add_eeg_ref=False, verbose=False,
+                          preload=True)
+    events = read_events(event_name)
+    picks = pick_types(raw.info, meg=False, eeg=True, stim=False,
+                       ecg=False, eog=False,
+                       exclude='bads')[::8]
+    return raw, events, picks
+
+
+@requires_sklearn
+def test_xdawntransformer_fit():
+    """Test Xdawn fit."""
+    # get data
+    raw, events, picks = _get_data()
+    epochs = Epochs(raw, events, event_id, tmin, tmax, picks=picks,
+                    preload=True, baseline=None, verbose=False)
+    X = epochs._data
+    y = epochs.events[:, -1]
+    print(y)
+    # =========== Basic Fit test =================
+    # test base xdawn
+    xd = XdawnTransformer(n_components=2, signal_cov=None, reg=None)
```

Comment:
if default parameter, don't explicit them (this will automatically ensure that others don't modify default params)